### ISSUE TITLE
Add ability to analyze in VS Code while project is open in Understand or vice-versa

### ISSUE THREAD

**Issue by robbycbennett** [stinb/UnderstandForVSCode#12]
Error message when trying to analyze while using Understand for VS Code:
![The Analysis cannot be performed because the database is locked or read-only.
Last locked by process ID: 7776](https://github.com/stinb/UnderstandForVSCode/assets/22121365/80fce90a-57af-490c-ba44-f4d92fb5c6c0)

Additionally, VS Code has no error message like in the screenshot.

**Comment by jordancolbeth** [comment]
Userver is supposed to release the license on close, but this Understand error is persisting even after the folder/workspace is closed in VS Code. Had to delete and recreate the project to be able to reanalyze in Understand.

**Comment by RobbyCBennett** [comment]
That happens because userver isn't getting or processing requests/notifications from the client. It ignores the shutdown request and therefore it doesn't release the license.

**Comment by groke** [comment]
@dougPollard Is this even possible to do before Rob spends time looking at it? Would the Understand go into read-only mode?

**Comment by DougPollard** [comment]
No, it's not currently possible, as described. I assume the parse failed at write, which on a long analyze can be confusing and can cause you to think (and write) that it's working. I solved this with the mcp server. At the time I imagined generalizing it to non-mcp cases, but that doesn't even appear on my todo list any where right now.

The simplest solution today would be to literally use the mcp mechanism unchanged. If that's what you want let me know more details and I can provide a solution or modify mcp to support it more directly for you.

**Comment by groke** [comment]
For now I'm adding this to @DougPollard 's backlog to add the framework for the non-mcp cases being able to request reparses. No hurry though.
